network/player.go

```diff
@@ -35,16 +38,56 @@ func ConvertToGetPlayerResponse(p model.Player) GetPlayerResponse {
 			ID:   p.ID,
 			Name: p.Name,
 		},
-		Games: convertToParticipatingGames(p.Games),
 	}
 }
 
-func convertToParticipatingGames(mgs map[model.GameID]model.PlayerColor) map[model.GameID]string {
-	games := make(map[model.GameID]string, len(mgs))
-	for g, c := range mgs {
-		games[g] = c.String()
+type ActiveGame struct {
+	PlayerNamesByID  map[model.PlayerID]string `json:"players"`
+	PlayerColorsByID map[model.PlayerID]string `json:"colors"`
+	Created          time.Time                 `json:"created"`
+	LastMove         time.Time                 `json:"lastMove"`
+}
+
+type GetActiveGamesForPlayerResponse struct {
+	Player      Player                      `json:"player"`
+	ActiveGames map[model.GameID]ActiveGame `json:"activeGames"`
+}
+
+func ConvertToGetActiveGamesForPlayerResponse(
+	p model.Player,
+	games map[model.GameID]model.Game,
+) GetActiveGamesForPlayerResponse {
+
+	return GetActiveGamesForPlayerResponse{
+		Player: Player{
+			ID:   p.ID,
+			Name: p.Name,
+		},
+		ActiveGames: convertToParticipatingGames(p, games),
+	}
+}
+
+func convertToParticipatingGames(p model.Player, games map[model.GameID]model.Game) map[model.GameID]ActiveGame {
+	res := make(map[model.GameID]ActiveGame, len(p.Games))
+	for gID := range p.Games {
+		if mg, ok := games[gID]; ok {
+			res[gID] = getActiveGame(mg)
+		}
+	}
```

Comment:
Okay so here
https://github.com/joshprzybyszewski/cribbage/blob/016f632f294433b325bbbad581ace1165c300b03/server/server.go#L333-L344
we're passing in a map of games that _we know_ this player is a part of because we made it by iterating over `p.Games` and grabbing those from persistence. What are we accomplishing by iterating over `p.Games` again here then? I am confusion, @joshprzybyszewski explain!!!

Reply:
The `Games` on the player struct is currently just a map of id to player color, not `model.Game`

https://github.com/joshprzybyszewski/cribbage/blob/892fbf53fcbdcd3d5adb6cc8c7920b5cc477faba/model/model.go#L91

Reply:
But you're populating a `map[model.GameID]model.Game` in `server.go` by iterating through `p.Games`. Then you pass `p` and `games` into `network.ConvertToGetActiveGamesForPlayerResponse` where you iterate through `p.Games` and check if each is contained in `games` - won't `ok` _always_ be true?

Reply:
ah -- I see what you mean. I'm a proponent for "don't trust input params". If you're given something, you should always verify/validate that it meets your requirements. Just because that codepath exists today, doesn't mean it will always exist.

Reply:
Okay. If the number of games a player is in grows, this could become a lot, but since were just reassigning I guess it would take a lot of games to be problematic. 👍 thanks for the explanation